src/main/drivers/usb_msc_common.c

```diff
@@ -68,11 +68,20 @@ void mscInit(void)
 
 bool mscCheckBoot(void)
 {
-    const uint32_t bootModeRequest = persistentObjectRead(PERSISTENT_OBJECT_RESET_REASON);
-    return bootModeRequest == RESET_MSC_REQUEST;
-    // Note that we can't clear the persisent object after checking here. This is because
-    // this function is called multiple times during initialization. So we clear on a reset
-    // out of MSC mode.
+    static bool firstCheck = true;
+    static bool mscMode;
+
+    if (firstCheck) {
+        // Cache the bootup value of RESET_MSC_REQUEST
+        const uint32_t bootModeRequest = persistentObjectRead(PERSISTENT_OBJECT_RESET_REASON);
+        mscMode = (bootModeRequest == RESET_MSC_REQUEST);
+        // Ensure the next reset is to the configurator as the H7 processor retains the RTC value so
+        // a brief interruption of power is not enough to switch out of MSC mode
+        persistentObjectWrite(PERSISTENT_OBJECT_RESET_REASON, RESET_NONE);
```

Comment:
Couple of things here:
- in general, a `bool checkX()` function should not have a side effect, as this will catch users of it unaware. If anything, rename to `mscCheckBootAndReset()`;
- should the reset of the reset reason only happen if it is an MSC request? In other cases the reason might still be needed at a later stage.

Reply:
1. Done
2. The other cases are interpreted by the bootloader, but you make a good point. Fixed.